signer/services_test.go

```diff
@@ -17,29 +17,37 @@ import (
 	tmlog "github.com/tendermint/tendermint/libs/log"
 	tmservice "github.com/tendermint/tendermint/libs/service"
 
+	fork "github.com/kraken-hpc/go-fork"
 	"github.com/stretchr/testify/require"
 )
 
-func TestIsRunning(t *testing.T) {
-	if runtime.GOOS != "linux" {
-		t.Skip("test only valid on Linux")
-	}
-
-	homeDir := t.TempDir()
-	pidFilePath := filepath.Join(homeDir, "horcrux.pid")
+func init() {
+	fork.RegisterFunc("child", mockHorcruxChildProcess)
+	fork.Init()
+}
 
-	// this child process exists for linux, but not macOS
-	// TODO: make this test work with macOS, maybe spawn child process or dummy horcrux process.
-	pid := os.Getpid() + 1
-	err := os.WriteFile(
+func mockHorcruxChildProcess(pidFilePath string) {
+	_ = os.WriteFile(
 		pidFilePath,
-		[]byte(fmt.Sprintf("%d\n", pid)),
+		[]byte(fmt.Sprintf("%d\n", os.Getpid())),
 		0600,
 	)
-	require.NoError(t, err, "error writing pid file")
+}
+
+func TestIsRunning(t *testing.T) {
+	homeDir := t.TempDir()
+	pidFilePath := filepath.Join(homeDir, "horcrux.pid")
+
+	err := fork.Fork("child", pidFilePath)
```

Comment:
Comment what is going on with this fork package/function. It looks like this might only be used in test? It's an unfamiliar package.

Reply:
comment added